src/main/src/test/java/org/geoserver/security/GeoServerSecurityManagerTest.java

```diff
@@ -188,4 +189,25 @@ public void testWebLoginChainSessionCreation() throws Exception {
             config.getFilterChain().getRequestChainByName(GeoServerSecurityFilterChain.WEB_LOGIN_CHAIN_NAME);
         assertTrue(chain.isAllowSessionCreation());
     }
+    
+    @Test public void testGeoServerEnvParametrization() throws Exception {
+        GeoServerSecurityManager secMgr = getSecurityManager();
+        SecurityManagerConfig config = secMgr.loadSecurityConfig();
+        String oldRoleServiceName = config.getRoleServiceName();
+        
+        try {
+            if (GeoServerEnvironment.ALLOW_ENV_PARAMETRIZATION) {
+                System.setProperty("TEST_SYS_PROPERTY", oldRoleServiceName);
```

Comment:
I'd suggest to clear the property in finally, might create accidental coupling with other tests